hydradx/model/amm/arbitrage_agent.py

```diff
@@ -249,32 +358,49 @@ def calculate_arb_amount_ask(
         i += 1
         if max_iters is not None and i >= max_iters:
             break
-
-    return amt_low
-
-
-def execute_arb(state, cex, agent, all_swaps, buffer=0.0):
-    for tkn_pair in all_swaps:
-        swaps = all_swaps[tkn_pair]
-
-        for swap in swaps:
-            tkn_buy_dex = swap['dex']['buy_asset']
-            tkn_sell_dex = swap['dex']['sell_asset']
-            tkn_buy_cex = swap['cex']['buy_asset']
-            tkn_sell_cex = swap['cex']['sell_asset']
-
-            if swap['dex']['trade'] == 'buy' and swap['cex']['trade'] == 'sell':
-                # omnipool leg
-                state.swap(agent, tkn_buy=tkn_buy_dex, tkn_sell=tkn_sell_dex, buy_quantity=swap['dex']['amount'])
-                # CEX leg
-                cex.swap(agent, tkn_buy=tkn_buy_cex, tkn_sell=tkn_sell_cex, sell_quantity=swap['cex']['amount'])
-            elif swap['dex']['trade'] == 'sell' and swap['cex']['trade'] == 'buy':
-                # omnipool leg
-                state.swap(agent, tkn_buy=tkn_buy_dex, tkn_sell=tkn_sell_dex, sell_quantity=swap['dex']['amount'])
-                # CEX leg
-                cex.swap(agent, tkn_buy=tkn_buy_cex, tkn_sell=tkn_sell_cex, buy_quantity=swap['cex']['amount'])
-            else:
-                raise
+    if amt_low == min_amt:
+        return 0
+    else:
+        return amt_low
+
+
+def execute_arb(state, cex_dict, agent, all_swaps):
```

Comment:
I'll have to make changes to this in my merge, because in re-routing trades I break the symmetry between 'cex' and 'dex' trades. But there's nothing wrong with it as is.